### DISCUSSION TITLE
Bekomme die SW nicht auf das Board

### DISCUSSION THREAD

**Discussion by jenskapp** [dewenni/ESP_Buderus_KM271#46] [Q&A]
Hi, 
ich nutze das Board von Daniel aktuell noch mit HomeAssistant - das funktioniert super. Mein Ökosystem ist aber iobroker. Ich hab die SW hier gefunden - super Artbeit :-)
Ich wollte diese jetzt testen, allerdings hab ich ein Problem: Ich bekomme die SW nicht geflasht.
Flashvorgang hab ich über die aktuelle Version vom espressif tool gestartet. Verwendet hab ich sowohl das neuste Release als auch die 3 davor.
--> no xmc flash detected

ESPHome bekomme ich problemlos über den selben Weg aufgespielt.
Habt ihr eine Idee woran das liegen kann?

**Comment by dewenni** [comment]
Das mit dem Flashen und den Updates ist echt komisch.
Nur zur Sicherheit, du hast schon immer die Variante genommen wo im Dateinamen ...esp32_flash... steht, oder?

ich habe es gerade nochmal mit der v3.2.4 getestet und ich konnte es problemlos aufspielen.
Ich nutze allerdings die Kommandozeile auf einem MAC

```
esptool.py -p /dev/cu.usbserial-0001 write_flash 0x00 buderus_km271_esp32_flash_v3.2.4.bin
esptool.py v4.6.2
Serial port /dev/cu.usbserial-0001
Connecting.....................................
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting...
Detecting chip type... ESP32
Chip is ESP32-D0WD (revision v1.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: c4:de:e2:a0:e4:f4
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Flash will be erased from 0x00000000 to 0x0011dfff...
Compressed 1167456 bytes to 705257...
Wrote 1167456 bytes (705257 compressed) at 0x00000000 in 62.4 seconds (effective 149.6 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
```

**Comment by dewenni** [comment]
Wenn es dir möglich ist, könntest du Visual Studio Code mit PlatformIO installieren und das Programm per Download auf das Board aufspielen. Das sollte auf jeden Fall funktionieren.

**Comment by the78mole** [comment]
Da ich ständig mal Firmware flashen muss, und nicht jedes Mal dafür eine Entwicklungsumgebung oder ein doofes GUI starten möchte, habe ich mir ein PowerShell Script geschrieben und darin ist folgender Teil, um Dewennis binary zu flashen:
```powershell
$COMPORT = "COM3"
$BAUDRATE = 921600
$FWFILE = "buderus_km271_esp32_flash_v3.2.4.bin"
Write-Host "#########################################################"
Write-Host "Hier die Informationen zum Target:"	
C:\Users\daniel\Tools\esptool-v3.3.2-win64\esptool.exe --chip esp32 --port $COMPORT --baud $BAUDRATE flash_id

Write-Host "#########################################################"
Write-Host "Flashvorgang startet:"	
C:\Users\daniel\Tools\esptool-v3.3.2-win64\esptool.exe --chip esp32 --port $COMPORT --baud $BAUDRATE write_flash -z 0x0 $FWFILE
```
Dazu muss man natürlich das esptool installieren und die richtigen Pfade einsetzen. Mit der ersten esptool-Zeile prüfe ich die Verbindung zum Target und schaue kurz, ob der passende ESP32 vorhanden ist. In der zweiten esptool-Zeile passiert der eigentliche Flash-Vorgang. Das Binary muss im aktuellen Verzeichnis liegen.

Unter Linux ist das natürlich noch einfacher, weil man da das esptool (wenn man python/pip installiert hat) relativ einfach mittels `pip install esptool` installieren kann.

**Comment by jenskapp** [comment]
Hi, 

danke erstmal für die Tips.
@dewenni: ja es ist *flash.bin
@the78mole: hab mal deine Kommandozeile getestet - allerdings lädt er immer die Benutzeroberfläche. Die Flash Tools sind doch nur ein Download - oder hab ich da vielleicht schon was übersehen?

Hier noch die exakte Ausgabe:
```
C:\ESP32\buderus_km271_esp32_flash_v3.2.4.bin
1 kill
...............................................................
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
NO XMC flash  detected!
FLASH_CRYPT_CNT 0
ABS_DONE_0 False

is stub and send flash finish
```

**Reply by the78mole** [reply]
Da bin ich mit meinem Latein leider am Ende...

**Comment by dewenni** [comment]
@jenskapp wie ist der aktuelle Stand bei dir?
Läuft die Software mittlerweile?
Falls nicht, teste doch bitte nochmal das aktuelle Release
